compiler/resolution/initializerRules.cpp

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2004-2016 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "initializerRules.h"
+
+#include "astutil.h"
+#include "expr.h"
+#include "stmt.h"
+#include "stringutil.h"
+#include "symbol.h"
+#include "type.h"
+
+// Helper file for verifying the rules placed on initializers, and providing
+// the extra functionality associated with them.
+
+void temporaryInitializerFixup(CallExpr* call) {
```

Comment:
Sorry, this function is called from functionResolution and was inserted in my last PR.  I just moved it into the initializerRules file when I created it, since it seemed like a good idea to keep all this information in one place.  While it is very brittle, it hopefully won't live very long - the problem it is solving only occurs because we're in this interim state where constructors and initializers both exist.